core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -970,6 +1086,183 @@ private static long getDedupRowsWithAdditionalKeys(
         }
     }
 
+    private static void mergeRowGroup(
+            ReadOnlyObjList<? extends MemoryCR> oooColumns,
+            long sortedTimestampsAddr,
+            TableWriter tableWriter,
+            PartitionDecoder file,
+            int rowGroup,
+            int timestampIndex,
+            int timestampColumnType,
+            long mergeRangeLo,
+            long mergeRangeHi,
+            TableRecordMetadata tableWriterMetadata,
+            long srcOooBatchRowSize,
+            Path path
+    ) {
+        long timestampChunkPtr = file.decodeColumnChunk(rowGroup, timestampIndex, timestampColumnType);
+        long timestampDataPtr = PartitionDecoder.getChunkDataPtr(timestampChunkPtr);
+        long rowGroupRowCount = PartitionDecoder.getRowGroupRowCount(timestampChunkPtr);
+
+        long mergeBatchRowCount = mergeRangeHi - mergeRangeLo + 1;
+        long mergeRowCount = mergeBatchRowCount + rowGroupRowCount;
+
+        final long timestampMergeIndexSize = mergeRowCount * TIMESTAMP_MERGE_ENTRY_BYTES;
+        final long timestampMergeIndexAddr = createMergeIndex(
+                timestampDataPtr,
+                sortedTimestampsAddr,
+                0,
+                rowGroupRowCount - 1,
+                mergeRangeLo,
+                mergeRangeHi,
+                timestampMergeIndexSize
+        );
+
+        ObjList<MemoryCARW> mergedMemory = new ObjList<>();
+        try (PartitionDescriptor partitionDescriptor = new BorrowedMemoryPartitionDescriptor()) {
```

Comment:
Please refactor to reuse `BorrowedMemoryPartitionDescriptor` instances

Reply:
refactored to reuse it for all row groups, allocated an instance per partition